File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // only one valid answer
        std::unordered_map<int, int> mymap;
        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];

            // if diff is already in map then return them.
            if (mymap.find(diff) != mymap.end()) {
                return {mymap[diff], i};
            }

            mymap[nums[i]] = i;
        }

        return {};
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)

        l, r = 0, n - 1

        while (l <= r):
            mid = (l + r) // 2
            curr = nums[mid]

            if curr == target:
                return mid
            if curr > target:
                r = mid - 1
            elif curr < target:
                l = mid + 1

        return -1